### ISSUE TITLE
customize defaultModalWidget and remove QR slider

### ISSUE THREAD

**Issue by karenpoveda29** [nextchapterstudio/walletconnect_qrcode_modal_dart#21]
Hi @3ph , I'm using the library in a project for work, I've been working on this task for days now and I need to personalize the entire modal. However, there are two things I haven't been able to change and I need to.

1. I need to modify the border radius of the modal. 
2. I'm not using the QR code functionality and I need to remove the slider but the farthest I could go was to delete the texts inside it as you see in the image.  

My question is: Is there a way to modify the defaultModalWidget further, maybe adding that option (border radius) in the copy with?. And also, is there currently a way to remove the QR slider o is it possible to add an option to do so if that part of the modal is not needed? 

I understand the library is called QR modal, but if you only need the connection without the QR code, it'd be great if you could hide/remove that element. 

Any solution for this?. Thanks

![image](https://user-images.githubusercontent.com/53981435/215877001-fb2ee3e2-f9a1-48de-9d0a-15cceed60d5f.png)

**Comment by karenpoveda29** [comment]
I know this may not be the best place to post a question but I found nowhere else to do it. Thanks in advance.

**Comment by 3ph** [comment]
Hi @karenpoveda29 - I've just released version 0.2.0 which allows you to customize the whole selector widget. You can now do for example:

```dart
 modalBuilder: (context, uri, callback, defaultModalWidget) {
   return defaultModalWidget.copyWith(
     cardShape: RoundedRectangleBorder(
       borderRadius: BorderRadius.circular(16),
     ),
     selectorBuilder: (context, defaultSelectorWidget) => Container(),
   );
}
```

I've also added `cardShape` parameter where you can adjust the border radius (see the example above).

Anything else you need?

**Comment by karenpoveda29** [comment]
@3ph Tom, you're the best! Thank you for replying so soon and being so helpful. It's amazing the work that you do. 

There is something else I wanted to ask, don't know if it's possible, though. You see, the original design I have to implement is the one I show in the picture. I understand that having the list is just a workaround to iOS restrictions and that for the user it's better to just have one button and that's it. That's why we added the search bar, so they could find the wallet easily. But I was asked to implement the same design (list) in android o.O

![image](https://user-images.githubusercontent.com/53981435/216669992-8770fa44-d056-44dd-8dde-727479c36429.png)


Since I pretty much created a whole different widget for the `defaultWalletListWidget`, I thought that I could use the same widget instead of the button (for android) but the `walletButtonBuilder` only receives the URI, so that didn't work. We want to display the wallets the user has on the phone and the ones that don't so they can download them. 

I guess my question is: As it is right now, is there a way to have the same implementation for iOS (list of wallets) and for android? Is that even possible? 

Again, thanks in advance.

**Comment by 3ph** [comment]
Hi @karenpoveda29 - I can try to add some kind of platform override. Let me think about it :).

**Comment by karenpoveda29** [comment]
hi @3ph, that would be amazing, thank for even taking the time. I really appreciate it.

**Comment by 3ph** [comment]
@karenpoveda29 I've pushed a new version which added `platformOverrides` parameter. You can now define which type (button, mobile list or desktop list) to be used on any of the 3 platforms. Corresponding builders will then be used. Please see README and code for an example.

**Comment by karenpoveda29** [comment]
Hi @3ph , thank you so much, I was really excited to see the new addition. However, I implemented it but didn't work. Some wallets won't open, others go to the wallet but don't show the connect message, and some do work but not consistently. When the wallets are loading in the modal, the following messages appear in the console. 

![image](https://user-images.githubusercontent.com/53981435/219801090-a31f2845-1be1-4af5-bbb2-c599ac8e936a.png)

I searched the error and some of the answers suggested adding the wallets package name to the `AndroidManifest.xml` file, so I did. I previously had rainbow and metamask and added the last three, but that didn't help, the wallets weren't removed from the error message nor they opened either. 

![image](https://user-images.githubusercontent.com/53981435/219801681-ab12879a-00f3-4126-bde8-c2c3c0f2cbac.png)

In terms of the code. I added this 
```
platformOverrides: const ModalWalletPlatformOverrides(
              android: ModalWalletType.listMobile,
            ),
```
and removed the `walletButtonBuilder`. I don't know if I'm missing something else in the implementation. I would appreciate any light on this.

**Comment by 3ph** [comment]
Hi @karenpoveda29 - you will have to use your own `onWalletTap` handler in the `ModalWalletListBuilder`. The default one likely won't work as it's never meant to work like this in Android (thus the disclaimer in the README).

If you want to use the deep links you will have to add `<queries>` for each schema you want to handle (have a look at the https://pub.dev/packages/url_launcher example). This is hardcoded in the manifest file so not flexible at all.

Universal links (app links) might work better but you might have to skip `canLaunchUrl` as it might return a false result. And it might still not work if the wallet app doesn't register the intent to handle these links. That's why there is a consensus to use `wc:` schema on Android and whichever wallet app can handle it would show on the system dialog.

You might want to focus on a small subset of wallets and handle these individually. In any case, it would mean a lot of research and testing...

**Comment by karenpoveda29** [comment]
Hi @3ph , thanks for your reply. We had to discard the idea of using the same modal for Android (at least for now) for the things you mentioned above. We had to move on with the functionality so maybe will work on this idea later. I wanted to thank you for your willingness to help and all your work. I have a couple more questions I'd like to ask if it's ok about the library but is there a place I can DM you so we don't keep this issue open unnecessarily?
